pkg/server/event_handler.go

```diff
@@ -80,7 +80,7 @@ const (
 
 // An interface for sending pmap event downstream.
 type Messenger interface {
-	Send(context.Context, *v1alpha1.PmapEvent) error
+	Send(context.Context, *v1alpha1.PmapEvent, error) error
```

Comment:
@sqin2019  Do you think it is a little bit weird that we have to modify the messenger interface to include error? Maybe it is better to change the pmap event protocol?

Reply:
Or Send(context.Context, *pubsub.Message) error, so that we can wrap the error in the message.

Reply:
I think either works fine here, do we have a preference here?

Reply:
On a second thought, since pmap event is a wrapper on the real data, and processing error is part of pmap handling. Let's change the pmap event proto instead.

Reply:
What do you all think?

Reply:
I agree, I will go ahead and start.